appintro/src/main/java/com/github/paolorotolo/appintro/AppIntroBase.java

```diff
@@ -451,11 +482,48 @@ private void handleSlideChanged(Fragment oldFragment, Fragment newFragment) {
 
     /**
      * Called after a new slide has been selected
+     * {@link #canChangeLastSlide(int)} is called to check if the last slide can be changed.
      *
      * @param position Position of the new selected slide
      */
     protected void onPageSelected(int position) {
-        // Empty method
+        LogHelper.d(TAG, "Selected slide number: " + (position + 1));
+        canChangeLastSlide(position);
+    }
+
+
+    /**
+     * Called by {@link #onPageSelected(int)} to check if the user has requested permissions on the previous slide.
+     * This method is required as a safety check because otherwise,
+     * the user might skip a slide containing permissions.
+     * <p>
+     * This method will send the user back to the first slide that contains permissions.
+     *
+     * @param position - The position that the user has changed to.
+     *                 <p>
+     *                 TODO(1): This method will require tweaking when switching to Viewpager2 as
+     *                 there is no RTL support in the current viewpager and that messes up the position of slides.
+     *                 <p>
+     *                 If the position issue is resolved in Viewpager2, simply remove the code inside if(isRtl()).
+     */
+    private void canChangeLastSlide(int position) {
+        if (isRtl()) {
+            if (permissionsPositions.contains(position + 2) && !requestedSlides.contains(position + 2)) {
+                pager.setCurrentItem(position + 1, true);
+                return;
+            }
+        } else {
+            if (permissionsPositions.contains(position) && !requestedSlides.contains(position)) {
+                pager.setCurrentItem(position - 1, true);
+                return;
+            }
+        }
+        if (permissionsPositions.contains(position + 1) && !requestedSlides.contains(position + 1)) {
```

Comment:
This should definitely not happen here but instead inside the `PagerOnPageChangeListener`

Reply:
Done